🐛 Logger does not set default values

**Fiber version**
github.com/gofiber/fiber v1.12.1

**Issue description**
I want to log with out statecode 200. 
So I add Next options to logger using ctx.Fasthttp.Response.StatusCode() and get panic.

**Code snippet**

```go
package main

import (
	"github.com/gofiber/fiber"
	"github.com/gofiber/fiber/middleware"
)

func main() {
	app := fiber.New()
	app.Use(middleware.LoggerWithConfig(middleware.LoggerConfig{
		Next: func(ctx *fiber.Ctx) bool {
			return ctx.Fasthttp.Response.StatusCode() != 200
		},
		Format:     "${time} feph ${method} ${path} - ${status} - ${latency}\nRequest : ${body}\n",
		TimeFormat: "2006-01-02T15:04:05-0700",
	}))
	app.Get("/", hello)
	app.Listen(3000)
}

func hello(c *fiber.Ctx) {
	c.Send("Hello, World 👋!")
}
```

**Panic message**

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x822f1b]

goroutine 23 [running]:
github.com/gofiber/fiber/middleware.LoggerWithConfig.func2(0xc00059a100)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.1/middleware/logger.go:201 +0x21b
github.com/gofiber/fiber.(*App).next(0xc00006a060, 0xc00059a100, 0xc00059a100)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.1/router.go:109 +0x1c2
github.com/gofiber/fiber.(*App).handler(0xc00006a060, 0xc000140500)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.1/router.go:125 +0x64
github.com/valyala/fasthttp.(*Server).serveConn(0xc000598000, 0x9a57e0, 0xc000010078, 0x0, 0x0)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/server.go:2162 +0x11bb
github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc000124280, 0xc00000e040)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/workerpool.go:223 +0xc0
github.com/valyala/fasthttp.(*workerPool).getCh.func1(0xc000124280, 0xc00000e040, 0x84e980, 0xc00000e040)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/workerpool.go:195 +0x35
created by github.com/valyala/fasthttp.(*workerPool).getCh
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/workerpool.go:194 +0x101
exit status 2
```

example accure same panic

```go
package main

import (
	"github.com/gofiber/fiber"
	"github.com/gofiber/fiber/middleware"
)

func main() {
	app := fiber.New()

	// Default
	app.Use(middleware.Logger())

	// Custom logging format
	app.Use(middleware.Logger("${method} - ${path}"))

	// Custom Config
	app.Use(middleware.LoggerWithConfig(middleware.LoggerConfig{
		Next: func(ctx *fiber.Ctx) bool {
			return ctx.Path() != "/private"
		},
		Format: "${method} - ${path}",
	}))

	app.Listen(4000)
}

```

```
GET http://localhost:4000/private
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x8231cb]

goroutine 37 [running]:
github.com/gofiber/fiber/middleware.LoggerWithConfig.func2(0xc000528000)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/middleware/logger.go:201 +0x21b
github.com/gofiber/fiber.(*App).next(0xc000102480, 0xc000528000, 0xaa26c3a5046)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/router.go:107 +0x176
github.com/gofiber/fiber.(*Ctx).Next(0xc000528000, 0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/ctx.go:619 +0x138
github.com/gofiber/fiber/middleware.LoggerWithConfig.func2(0xc000528000)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/middleware/logger.go:139 +0xbd
github.com/gofiber/fiber.(*App).next(0xc000102480, 0xc000528000, 0xaa26c3a4e52)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/router.go:107 +0x176
github.com/gofiber/fiber.(*Ctx).Next(0xc000528000, 0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/ctx.go:619 +0x138
github.com/gofiber/fiber/middleware.LoggerWithConfig.func2(0xc000528000)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/middleware/logger.go:139 +0xbd
github.com/gofiber/fiber.(*App).next(0xc000102480, 0xc000528000, 0xc000528000)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/router.go:107 +0x176
github.com/gofiber/fiber.(*App).handler(0xc000102480, 0xc000520000)
        /go/pkg/mod/github.com/gofiber/fiber@v1.12.2/router.go:123 +0x64
github.com/valyala/fasthttp.(*Server).serveConn(0xc00010e6c0, 0x9a5660, 0xc000010030, 0x0, 0x0)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/server.go:2162 +0x11bb
github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc00007ed20, 0xc00000e040)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/workerpool.go:223 +0xc0
github.com/valyala/fasthttp.(*workerPool).getCh.func1(0xc00007ed20, 0xc00000e040, 0x84e980, 0xc00000e040)
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/workerpool.go:195 +0x35
created by github.com/valyala/fasthttp.(*workerPool).getCh
        /go/pkg/mod/github.com/valyala/fasthttp@v1.14.0/workerpool.go:194 +0x101
exit status 2
```

--- a/middleware/logger.go
+++ b/middleware/logger.go
@@ -108,6 +108,19 @@ func Logger(format ...string) fiber.Handler {
 
 // LoggerWithConfig allows you to pass an CompressConfig
 func LoggerWithConfig(config LoggerConfig) fiber.Handler {
+	// Set config default values
+	if config.Format == "" {
+		config.Format = LoggerConfigDefault.Format
+	}
+
+	if config.TimeFormat == "" {
+		config.TimeFormat = LoggerConfigDefault.TimeFormat
+	}
+
+	if config.Output == nil {
+		config.Output = LoggerConfigDefault.Output
+	}
+
 	// Middleware settings
 	var mutex sync.RWMutex
 

--- a/middleware/logger_test.go
+++ b/middleware/logger_test.go
@@ -46,7 +46,7 @@ func Test_Middleware_Logger(t *testing.T) {
 
 }
 
-func Test_Middleware_Logger_WithDefaulFormat(t *testing.T) {
+func Test_Middleware_Logger_WithDefaultFormat(t *testing.T) {
 	expectedOutputPattern := regexp.MustCompile(`^\d{2}:\d{2}:\d{2} GET / - 0\.0\.0\.0 - 200 - \d+(\.\d+)?.{1,3}
 $`)
 	// fake output

